sdk/core/azure-core/inc/azure/core/datetime.hpp

```diff
@@ -324,6 +340,52 @@ namespace Core { namespace _internal {
      */
     ~PosixTimeConverter() = delete;
   };
+
+  /**
+   * @brief Provides additional methods for #Azure::DateTime.
+   *
+   */
+  class DateTimeExtensions final {
```

Comment:
This could be a private method.  Or if we want to expose it to customers then its a public method.  Why the complexity of a friend class?

Reply:
If it is a private method, then how would we call it from Core Tests, and from Identity's TokenCredentialImpl?
I'd rather not expose it to customers - no need for that, no need to commit to support it, and making no breaking changes.
So, `Core::_internal::DateTimeExtensions::Parse()` seems ideal to me - it is internal API that is shared between SDK libraries, but it is not available to the public. And I don't know how to achieve that without friends.

Reply:
BTW, the new iteration no longer needs to use friends.